File: 0078-subsets/0078-subsets.java
class Solution {
    public List<List<Integer>> subsets(int[] nums) {
        List<List<Integer>> subsets = new ArrayList<>();
        subsets.add(new ArrayList<>());
        for (int i : nums) {
           List<List<Integer>> tempSubsets = new ArrayList<>();
            for (List<Integer> subset : subsets) {
                 List<Integer> newTempSubsets = new ArrayList<>(subset);
                // temp=;
                newTempSubsets.add(i);
                tempSubsets.add(newTempSubsets);
            }
            subsets.addAll(tempSubsets);
        }
        return subsets;
    }
}

File: 0078-subsets/0078-subsets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        ans = []
        def helper(path,idx):
            ans.append(path[:])
            for i in range(idx,len(nums)):
                path.append(nums[i])
                helper(path,i+1)
                path.pop()
        helper([],0)
        return ans

        